<a href="https://colab.research.google.com/github/Shrinjita/21AIC401T_Healthcare_Analytics_Case_Study/blob/main/ShrinjitaPaul_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

DOWLOADING KAGGLE FILE

In [1]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("alexteboul/diabetes-health-indicators-dataset")

print("Path to dataset files:", path)

100%|██████████| 6.03M/6.03M [00:00<00:00, 112MB/s]

Extracting files...


Path to dataset files: /root/.cache/kagglehub/datasets/alexteboul/diabetes-health-indicators-dataset/versions/1


In [2]:
# Import the necessary library
import pandas as pd

# Load the Dataset

In [3]:
# Since the exact filename might be tricky, let's first list the files in that directory to be sure
import os
dataset_path = "/root/.cache/kagglehub/datasets/alexteboul/diabetes-health-indicators-dataset/versions/1"
print("Files in directory:", os.listdir(dataset_path))

Files in directory: ['diabetes_binary_health_indicators_BRFSS2015.csv', 'diabetes_012_health_indicators_BRFSS2015.csv', 'diabetes_binary_5050split_health_indicators_BRFSS2015.csv']


In [4]:
file_name = "diabetes_binary_health_indicators_BRFSS2015.csv"  # The main file
full_path = f"{dataset_path}/{file_name}"

In [5]:
# Load the dataset into a Pandas DataFrame
df = pd.read_csv(full_path)

# Confirm it's loaded by checking the first 5 rows
print("Dataset loaded successfully! First 5 rows:")
df.head()

Dataset loaded successfully! First 5 rows:


Diabetes_binary  HighBP  HighChol  CholCheck   BMI  Smoker  Stroke  \
0              0.0     1.0       1.0        1.0  40.0     1.0     0.0   
1              0.0     0.0       0.0        0.0  25.0     1.0     0.0   
2              0.0     1.0       1.0        1.0  28.0     0.0     0.0   
3              0.0     1.0       0.0        1.0  27.0     0.0     0.0   
4              0.0     1.0       1.0        1.0  24.0     0.0     0.0   

   HeartDiseaseorAttack  PhysActivity  Fruits  ...  AnyHealthcare  \
0                   0.0           0.0     0.0  ...            1.0   
1                   0.0           1.0     0.0  ...            0.0   
2                   0.0           0.0     1.0  ...            1.0   
3                   0.0           1.0     1.0  ...            1.0   
4                   0.0           1.0     1.0  ...            1.0   

   NoDocbcCost  GenHlth  MentHlth  PhysHlth  DiffWalk  Sex   Age  Education  \
0          0.0      5.0      18.0      15.0       1.0  0.0   9.0        4.0   
1          1.0      3.0       0.0       0.0       0.0  0.0   7.0        6.0   
2          1.0      5.0      30.0      30.0       1.0  0.0   9.0        4.0   
3          0.0      2.0       0.0       0.0       0.0  0.0  11.0        3.0   
4          0.0      2.0       3.0       0.0       0.0  0.0  11.0        5.0   

   Income  
0     3.0  
1     1.0  
2     8.0  
3     6.0  
4     4.0  

[5 rows x 22 columns]

# EDA

In [6]:
# Check the dimensions of the dataset (rows, columns)
print("Dataset Shape:", df.shape)

# Get basic info: data types and non-null counts
df.info()

# Generate descriptive statistics for all columns
df.describe().T

Dataset Shape: (253680, 22)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 253680 entries, 0 to 253679
Data columns (total 22 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   Diabetes_binary       253680 non-null  float64
 1   HighBP                253680 non-null  float64
 2   HighChol              253680 non-null  float64
 3   CholCheck             253680 non-null  float64
 4   BMI                   253680 non-null  float64
 5   Smoker                253680 non-null  float64
 6   Stroke                253680 non-null  float64
 7   HeartDiseaseorAttack  253680 non-null  float64
 8   PhysActivity          253680 non-null  float64
 9   Fruits                253680 non-null  float64
 10  Veggies               253680 non-null  float64
 11  HvyAlcoholConsump     253680 non-null  float64
 12  AnyHealthcare         253680 non-null  float64
 13  NoDocbcCost           253680 non-null  float64
 14  GenHlth               25

count       mean       std   min   25%   50%   75%  \
Diabetes_binary       253680.0   0.139333  0.346294   0.0   0.0   0.0   0.0   
HighBP                253680.0   0.429001  0.494934   0.0   0.0   0.0   1.0   
HighChol              253680.0   0.424121  0.494210   0.0   0.0   0.0   1.0   
CholCheck             253680.0   0.962670  0.189571   0.0   1.0   1.0   1.0   
BMI                   253680.0  28.382364  6.608694  12.0  24.0  27.0  31.0   
Smoker                253680.0   0.443169  0.496761   0.0   0.0   0.0   1.0   
Stroke                253680.0   0.040571  0.197294   0.0   0.0   0.0   0.0   
HeartDiseaseorAttack  253680.0   0.094186  0.292087   0.0   0.0   0.0   0.0   
PhysActivity          253680.0   0.756544  0.429169   0.0   1.0   1.0   1.0   
Fruits                253680.0   0.634256  0.481639   0.0   0.0   1.0   1.0   
Veggies               253680.0   0.811420  0.391175   0.0   1.0   1.0   1.0   
HvyAlcoholConsump     253680.0   0.056197  0.230302   0.0   0.0   0.0   0.0   
AnyHealthcare         253680.0   0.951053  0.215759   0.0   1.0   1.0   1.0   
NoDocbcCost           253680.0   0.084177  0.277654   0.0   0.0   0.0   0.0   
GenHlth               253680.0   2.511392  1.068477   1.0   2.0   2.0   3.0   
MentHlth              253680.0   3.184772  7.412847   0.0   0.0   0.0   2.0   
PhysHlth              253680.0   4.242081  8.717951   0.0   0.0   0.0   3.0   
DiffWalk              253680.0   0.168224  0.374066   0.0   0.0   0.0   0.0   
Sex                   253680.0   0.440342  0.496429   0.0   0.0   0.0   1.0   
Age                   253680.0   8.032119  3.054220   1.0   6.0   8.0  10.0   
Education             253680.0   5.050434  0.985774   1.0   4.0   5.0   6.0   
Income                253680.0   6.053875  2.071148   1.0   5.0   7.0   8.0   

                       max  
Diabetes_binary        1.0  
HighBP                 1.0  
HighChol               1.0  
CholCheck              1.0  
BMI                   98.0  
Smoker                 1.0  
Stroke                 1.0  
HeartDiseaseorAttack   1.0  
PhysActivity           1.0  
Fruits                 1.0  
Veggies                1.0  
HvyAlcoholConsump      1.0  
AnyHealthcare          1.0  
NoDocbcCost            1.0  
GenHlth                5.0  
MentHlth              30.0  
PhysHlth              30.0  
DiffWalk               1.0  
Sex                    1.0  
Age                   13.0  
Education              6.0  
Income                 8.0

# Data Cleaning & Preprocessing

In [7]:
# Define a function to recode the age groups
def recode_age(age_code):
    if age_code <= 3:
        return 'Young Adult (18-39)'
    elif age_code <= 7:
        return 'Middle Aged (40-59)'
    else:
        return 'Senior (60+)'

# Apply the function to create a new 'Age_Group' column
df['Age_Group'] = df['Age'].apply(recode_age)

# Verify the recoding was successful
print(df['Age_Group'].value_counts())

Age_Group
Senior (60+)           153146
Middle Aged (40-59)     76113
Young Adult (18-39)     24421
Name: count, dtype: int64


Check for Missing Values

In [8]:
# Check for any missing values in the entire dataset
print("Missing values in each column:")
print(df.isnull().sum())

Missing values in each column:
Diabetes_binary         0
HighBP                  0
HighChol                0
CholCheck               0
BMI                     0
Smoker                  0
Stroke                  0
HeartDiseaseorAttack    0
PhysActivity            0
Fruits                  0
Veggies                 0
HvyAlcoholConsump       0
AnyHealthcare           0
NoDocbcCost             0
GenHlth                 0
MentHlth                0
PhysHlth                0
DiffWalk                0
Sex                     0
Age                     0
Education               0
Income                  0
Age_Group               0
dtype: int64


# One-Sample T-Test
[In the Body Mass Index (BMI) scale, established by the World Health Organization (WHO) and other health bodies, the number 25 is a critical clinical threshold.]

Test if the mean BMI is different from 25.

In [9]:
from scipy import stats

# Perform one-sample t-test
t_statistic, p_value = stats.ttest_1samp(df['BMI'], popmean=25)

print("--- One-Sample T-Test Results (BMI vs. 25) ---")
print(f"T-Statistic: {t_statistic:.4f}")
print(f"P-Value: {p_value:.4f}")
print(f"Mean BMI: {df['BMI'].mean():.2f}")
if p_value < 0.05:
    print("Conclusion: Reject the null hypothesis. The mean BMI is significantly different from 25.")
else:
    print("Conclusion: Fail to reject the null hypothesis. There is no significant difference.")

--- One-Sample T-Test Results (BMI vs. 25) ---
T-Statistic: 257.7791
P-Value: 0.0000
Mean BMI: 28.38
Conclusion: Reject the null hypothesis. The mean BMI is significantly different from 25.


Null Hypothesis (H₀): μ = 25

Alternative Hypothesis (H₁): μ ≠ 25

Significance Level (α): 0.05

Results:

μ (Hypothesized Population Mean): 25

t-statistic (Calculated): t_calculated = 257.78

Critical t-value (from table): t_critical ≈ ±1.96 (For a large sample size, α=0.05, two-tailed)

p-value: p_value = 0.0000

Decision Rule:

Reject H₀ if: |t_calculated| > t_critical OR p-value < α

Conclusion:
Since |257.78| > 1.96 and 0.0000 < 0.05, we reject the null hypothesis (H₀).

There is sufficient evidence at the 0.05 significance level to conclude that the true population mean BMI is significantly different from 25. The sample mean of 28.38 suggests it is significantly greater than the WHO healthy-weight threshold.

# B. Independent Two-Sample T-Test
Test if mean PhysHlth (poor health days) differs by Sex (0=Female, 1=Male).

In [10]:
# Separate the data into two groups
group_f = df[df['Sex'] == 0]['PhysHlth']  # Female
group_m = df[df['Sex'] == 1]['PhysHlth']  # Male

# Check equality of variances (Assumption for t-test)
lev_stat, lev_p = stats.levene(group_f, group_m)
print(f"\n--- Levene's Test for Equality of Variances ---")
print(f"W-statistic: {lev_stat:.3f}, p-value: {lev_p:.4f}")

# Perform the t-test. Use equal_var=False if Levene's p-value is significant (<0.05)
t_stat, ttest_p = stats.ttest_ind(group_f, group_m, equal_var=(lev_p > 0.05))

print("\n--- Two-Sample T-Test Results (PhysHlth by Sex) ---")
print(f"T-Statistic: {t_stat:.4f}")
print(f"P-Value: {ttest_p:.4f}")
print(f"Mean PhysHlth - Female: {group_f.mean():.2f}")
print(f"Mean PhysHlth - Male: {group_m.mean():.2f}")
if ttest_p < 0.05:
    print("Conclusion: Reject the null hypothesis. There is a significant difference in means.")
else:
    print("Conclusion: Fail to reject the null hypothesis. There is no significant difference.")


--- Levene's Test for Equality of Variances ---
W-statistic: 472.913, p-value: 0.0000

--- Two-Sample T-Test Results (PhysHlth by Sex) ---
T-Statistic: 21.8768
P-Value: 0.0000
Mean PhysHlth - Female: 4.58
Mean PhysHlth - Male: 3.82
Conclusion: Reject the null hypothesis. There is a significant difference in means.


Groups: Female (Sex = 0) vs. Male (Sex = 1)

Null Hypothesis (H₀): μ_female = μ_male

Alternative Hypothesis (H₁): μ_female ≠ μ_male

Significance Level (α): 0.05

Assumption Check (Levene's Test for Equality of Variances):

W-statistic = 472.913

p-value = 0.0000

Since p-value < 0.05, we reject the null hypothesis of equal variances. Thus, we use Welch's t-test (equal_var=False).

Results:

t-statistic (Calculated): t_calculated = 21.8768

Critical t-value (from table): t_critical ≈ ±1.96 (For a large sample size, α=0.05, two-tailed)

p-value: p_value = 0.0000

Sample Mean - Female: x̄_female = 4.58

Sample Mean - Male: x̄_male = 3.82

Decision Rule:

Reject H₀ if |t_calculated| > t_critical or p-value < α

Conclusion:
Since |21.8768| > 1.96 and 0.0000 < 0.05, we reject the null hypothesis (H₀). There is sufficient evidence at the 0.05 significance level to conclude that there is a statistically significant difference in the mean number of poor physical health days between females and males. The sample means indicate that females report more poor physical health days on average (4.58) compared to males (3.82).

# One-Way ANOVA with Post-Hoc Tukey Test
Test if mean BMI differs across your new Age_Group variable.

In [11]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Separate the data into the three age groups
group_young = df[df['Age_Group'] == 'Young Adult (18-39)']['BMI']
group_mid = df[df['Age_Group'] == 'Middle Aged (40-59)']['BMI']
group_senior = df[df['Age_Group'] == 'Senior (60+)']['BMI']

# Perform One-Way ANOVA
f_stat, p_value = stats.f_oneway(group_young, group_mid, group_senior)

print("\n--- One-Way ANOVA Results (BMI by Age Group) ---")
print(f"F-Statistic: {f_stat:.4f}")
print(f"P-Value: {p_value:.4f}")

if p_value < 0.05:
    print("Conclusion: Reject the null hypothesis. At least one group mean is significantly different.")

    # Perform Tukey's HSD Post-Hoc Test since ANOVA was significant
    print("\n--- Tukey's HSD Post-Hoc Test ---")
    tukey_results = pairwise_tukeyhsd(endog=df['BMI'], groups=df['Age_Group'], alpha=0.05)
    print(tukey_results)

else:
    print("Conclusion: Fail to reject the null hypothesis. No significant differences found.")


--- One-Way ANOVA Results (BMI by Age Group) ---
F-Statistic: 415.3345
P-Value: 0.0000
Conclusion: Reject the null hypothesis. At least one group mean is significantly different.

--- Tukey's HSD Post-Hoc Test ---
             Multiple Comparison of Means - Tukey HSD, FWER=0.05             
       group1              group2       meandiff p-adj  lower   upper  reject
-----------------------------------------------------------------------------
Middle Aged (40-59)        Senior (60+)  -0.6392   0.0 -0.7078 -0.5707   True
Middle Aged (40-59) Young Adult (18-39)  -1.2619   0.0 -1.3757 -1.1482   True
       Senior (60+) Young Adult (18-39)  -0.6227   0.0 -0.7292 -0.5161   True
-----------------------------------------------------------------------------


Groups: Young Adult (18-39), Middle Aged (40-59), Senior (60+)

Null Hypothesis (H₀): μ_young = μ_middle = μ_senior (All group means are equal)

Alternative Hypothesis (H₁): At least one group mean is significantly different

Significance Level (α): 0.05

Results:

F-statistic (Calculated): F_calculated = 415.3345

Critical F-value (from table): F_critical ≈ 3.00 (Approximate value for df_between=2, df_within=large, α=0.05)

p-value: p_value = 0.0000

Decision Rule for ANOVA:

Reject H₀ if F_calculated > F_critical or p-value < α

Conclusion for ANOVA:
Since F_calculated (415.3345) > F_critical (3.00) and p_value (0.0000) < α (0.05), we reject the null hypothesis (H₀). There is sufficient evidence at the 0.05 significance level to conclude that at least one age group has a significantly different mean BMI.

In [13]:
# Create a Pandas Excel writer object
with pd.ExcelWriter('YourName_CaseStudy.xlsx') as writer: # <- Change the name!

    # Save the main cleaned DataFrame to a sheet
    df.to_excel(writer, sheet_name='Cleaned_Data', index=False)

    # Create a small DataFrame for the README sheet explaining the key variables
    readme_data = {
        'Variable Name': ['BMI', 'PhysHlth', 'Sex', 'Age', 'Age_Group'],
        'Description': [
            'Body Mass Index (Continuous)',
            'Number of days physical health was not good (last 30 days)',
            'Biological sex (0 = Female, 1 = Male)',
            'Original age category (1-13)',
            'Recoded age group for ANOVA (Young Adult, Middle Aged, Senior)'
        ]
    }
    readme_df = pd.DataFrame(readme_data)

    # Save the README DataFrame to another sheet
    readme_df.to_excel(writer, sheet_name='README', index=False)

print("Excel file 'ShrinjitaPaul_CaseStudy.xlsx' has been created successfully!")

Excel file 'ShrinjitaPaul_CaseStudy.xlsx' has been created successfully!


Since the ANOVA was significant, we conducted Tukey's HSD test to identify which specific group means differ.

Results of Tukey's HSD Test:

All pairwise comparisons are statistically significant (adjusted p-value < 0.05 for all pairs).

Mean Differences:

Middle Aged vs. Senior: -0.6392 (Senior has higher BMI)

Middle Aged vs. Young Adult: -1.2619 (Young Adult has higher BMI)

Senior vs. Young Adult: -0.6227 (Young Adult has higher BMI)

Conclusion:
The post-hoc analysis reveals that all age groups have significantly different mean BMI values. Specifically:

Young Adults (18-39) have a significantly higher mean BMI than both Middle Aged (40-59) and Seniors (60+).

Seniors (60+) have a significantly higher mean BMI than Middle Aged (40-59) individuals.

This suggests that age group is a significant factor in BMI variation, with Young Adults exhibiting the highest BMI on average, followed by Seniors, and then Middle Aged individuals.